internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1453,9 +1455,18 @@ func (r *ClusterDeploymentsReconciler) getReleaseImage(
 
 	releaseImage, err := r.VersionsHandler.GetReleaseImageByURL(ctx, clusterImageSet.Spec.ReleaseImage, pullSecret)
 	if err != nil {
+		errMsgSuffix := ""
+		if r.MirrorRegistriesConfigBuilder.IsMirrorRegistriesConfigured() {
+			// There is a special error case where we should check to see if the supplied image is tag based
+			// if so then we need to provide the user with a hint as to what might be happening.
+			ref, err2 := imageReference.Parse(clusterImageSet.Spec.ReleaseImage)
+			if err2 == nil && len(ref.Tag) > 0 {
+				errMsgSuffix = "as the image is tag based and this can cause issues with oc, please take this into consideration"
```

Comment:
wondering if message is clear, @eranco74 @omertuc WDYT?

Reply:
Consider changing it to:
```suggestion
				errMsgSuffix = "as the image is tag based and this can cause issues with oc, please use image digest instead"
```

Reply:
`oc` is an implementation detail, our users should not be aware of it, so we should not mention it

Users also might not be familiar with pull-spec terminology such as "tag" so I would help them out with an example, by including the tag itself in the error message. So maybe something like:;

`fmt.Sprintf("Release image %q uses a tag (%q), this is usually unsupported in combination with mirror registries, please provide a digest-based image instead", clusterImageSet.Spec.ReleaseImage, ref.Tag)`

Feel free to drop "usually" if you think it's not a good fit, I'm not familiar with the technical details here so I'm not sure